File: python/teststack.py
import pytest
from stack import Stack

def test_push_pop():
    s = Stack()
    s.push(10)
    s.push(20)
    assert s.size() == 2
    assert s.pop() == 20
    assert s.top() == 10

def test_empty_operations():
    s = Stack()
    assert s.is_empty()
    assert s.pop() is None
    assert s.top() is None
    assert s.size() == 0

def test_clear():
    s = Stack()
    s.push(5)
    s.clear()
    assert s.is_empty()
    assert s.size() == 0

File: tests/teststack.cpp
#include <gtest/gtest.h>
#include "../include/stack.h"

TEST(StackTest, PushPopTop) {
    Stack s;
    EXPECT_TRUE(s.empty());
    s.push(10);
    s.push(20);
    EXPECT_EQ(s.size(), 2);
    EXPECT_EQ(s.top(), 20);
    s.pop();
    EXPECT_EQ(s.top(), 10);
}

TEST(StackTest, PrintAndClear) {
    Stack s;
    s.push(5);
    s.push(15);
    s.clear();
    EXPECT_TRUE(s.empty());
    EXPECT_EQ(s.size(), 0);
}

TEST(StackTest, EdgeCases) {
    Stack s;
    s.pop();  
    EXPECT_EQ(s.top(), 0);
    EXPECT_EQ(s.size(), 0);
    EXPECT_TRUE(s.empty());
}